tests/test_eic.cc

```diff
@@ -0,0 +1,269 @@
+
+    string line;
+    vector<double> vec;
+    vector< vector<double> > matrix;
+
+    while( std::getline(infile,line) ){
+        std::stringstream ss(line);
+        vector<double> row;
+        std::string data;
+        while (getline(ss,data,',') ){
+
+            //Handle lines that don't match the format
+            try{
+                row.push_back( stod(data) );
+            }
+            catch(const std::exception& e){
+             //It was a bad line, we'll toss the whole row   
+                continue;
+            }
+        }
+        //The variables are F_const, v_trans, v_long, electron density, force_trans, force_long
+        if (row.size()  == 6) matrix.push_back(row);
+    }    
+
+    //now that we've ingested the whole file,
+    // make cuts to get the range we want to test.
+
+    vector< vector<double> > out_matrix;
+
+    for(int i=0;i<matrix.size();i++){
+        vector<double> vec = matrix[i];
+        if(vec.size()>0){
+            if (vec[1] < 1){ //Pick a specific low v_transverse
+                out_matrix.push_back(vec);
+            }
+        }
+    }
+    
+    //Now we need to sort by v_long and return it to the
+    // comparison function
+    sortrows(out_matrix,2);
+    
+    return out_matrix;
+}
+
+//Parse the file generated in SetupModel and compare it to the 
+// 'golden' dataset that is stored in the data directory
+double CompareOutput(string filename_golden,string filename_test){
+
+    vector< vector<double> > golden = ReadCSV(filename_golden);
+    vector< vector<double> > test = ReadCSV(filename_test);
+
+    //Calculate the difference between these trends, so there's just
+    // one trend to fit    
+    vector<double> x_vec;
+    vector<double> y_vec;
+    vector<double> y_vec_g;
+    int n = 0;
+
+    for(int i = 0; i<golden.size(); i++){
+        
+        double v_long_g = golden[i][2];
+        double f_long_g = golden[i][5];
+     
+        //Find the matching point for the test dataset
+        // & add it to the fit vector. Our data is already
+        // sorted by v_long.
+        double min = 10000; //The closest point
+        int min_index = 0;
+        
+        for(int j = 0; j<test.size();j++){
+            double v_long_t = test[j][2];
+            //can't expect the sampled velocities to be exactly
+            // the same, but check that they're close
+
+            if(abs(v_long_t - v_long_g) < min){
+                min = abs(v_long_t - v_long_g);
+                min_index = j;
+            }
+        }
+
+        double f_long_t = test[min_index][5];
+        x_vec.push_back(test[min_index][2]);
+        y_vec.push_back(f_long_t - f_long_g);
+        y_vec_g.push_back(f_long_g);
+        n++;               
+    }
+
+    //Use the GSL library we've already introduced as a depencency
+    // to compare these trends. Initialize the inputs and outputs:
+    double cov00,cov01,cov11;
+    double c0,c1,sumsq;
+    
+    double x[n];
```

Comment:
Probably want

```cpp
double* x = x_vec.data(); 
```
You probably also want to ensure that `x_vec.size() == n`. I know the loop does this, but it's one of those things that if it breaks, things can go really wrong.

Reply:
OK. This construct might show up in a few other places (and maybe in the upstream source) but I can go through and make this change wherever it appears.

Reply:
I realized the call to [gsl_fit_linear](https://github.com/radiasoft/electroncooling/blob/d6fef3b4067cdacafc19c8e87dd9eead8cc0415a/tests/test_eic.cc#L224) should probably look like this:

```cpp
assert(x_vec.size() == y_vec.size());
gsl_fit_linear(
    x_vec.data(),
    xstride,
    y_vec.data(),
    ystride,
    x_vec.size(),
    &c0,
    &c1,
    &cov00,
    &cov01,
    &cov11,
    &sumsq,
);
```

Most of the earlier code could be eliminated, and the code is more reliable.

Reply:
I appreciate these comments, I'm still shaking a little rust off with C++. Do you want to verify that the changes you've suggested are satisfied, or should I click Resolve Conversation for each of these?

Reply:
C++ is an extremely complex language so being rusty is normal. ;-)

I'm fine with whatever you do. I was just noting some things. Thank you for considering my input!